invenio_pages/records/models.py

```diff
@@ -39,23 +45,85 @@ class PageModel(db.Model, Timestamp):
     template_name = db.Column(db.String(70), nullable=False)
     """Page template name."""
 
+    @classmethod
+    def create(self, data):
+        """Create a new page."""
+        try:
+            with db.session.begin_nested():
+                obj = PageModel(
+                    url=data["url"],
+                    title=data.get("title", ""),
+                    content=data.get("content", ""),
+                    description=data.get("description", ""),
+                    template_name=data["template_name"],
+                )
+                db.session.add(obj)
+
+            return obj
+        except IntegrityError:
```

Comment:
As discussed, it is probably better to raise the error when creating a page with an URL that already exists, and show to the user a nice error